### PR TITLE
Remove routing decorators in `test_application.py`

### PR DESCRIPTION
**Pull request by aminalaee**
Subset of #1481 drops usage of routing decorators in tests/test_application.py .

### CODE REVIEW

**Comment by aminalaee** [comment]
@tomchristie I've left this to fail on purpose because we're avoiding `add_middleware` method and `host` and `mount` decorators which will lack coverage. Should we treat these the same way and `no cover` them or leave them as they are and clean in another round of PRs?

**Comment by lovelydinosaur** [comment]
We should also `no pragma` those, yup.

docstrings on `host` and `mount` would make sense.

You can either bundle all that into this PR, or treat it separately. It probably doesn't matter that much.

**Comment by aminalaee** [comment]
Based on changes here, I'll double check if there are any `mount` or `host` or any other deprecated usages in the files already merged. Might need another PR to clean all of them.

File: starlette/applications.py [hunk 1]
```diff
@@ -107,7 +107,14 @@ def debug(self) -> bool:
...
     @debug.setter
-    def debug(self, value: bool) -> None:
+    def debug(self, value: bool) -> None:  # pragma: no cover
+        """
+        We no longer document this API, and its usage is discouraged.
+        Instead you should use the following approach:
+
+        app = Starlette(debug=True)
+        """
+
```
Comment (lovelydinosaur) [inline starlette/applications.py]: Not obvious that we want to change this. It's not in the "The following usages are now discouraged..." block. Can we keep it as-is, and discuss any changes there separately?

```suggestion
    def debug(self, value: bool) -> None:
```

Comment (aminalaee) [reply starlette/applications.py]: Reverted this change for later discussion.

**Review by lovelydinosaur — APPROVED** [review]

**Pull request merged**